src/renderers/webgl/WebGLProgram.js

```diff
@@ -786,13 +786,31 @@ function WebGLProgram( renderer, cacheKey, parameters, bindingStates ) {
 			const vertexErrors = getShaderErrors( gl, glVertexShader, 'vertex' );
 			const fragmentErrors = getShaderErrors( gl, glFragmentShader, 'fragment' );
 
-			console.error(
-				'THREE.WebGLProgram: Shader Error ' + gl.getError() + ' - ' +
-				'VALIDATE_STATUS ' + gl.getProgramParameter( program, gl.VALIDATE_STATUS ) + '\n\n' +
+			const glError = gl.getError();
+			const validateStatus = gl.getProgramParameter( program, gl.VALIDATE_STATUS );
+			const error = (
+				'THREE.WebGLProgram: Shader Error ' + glError + ' - ' +
+				'VALIDATE_STATUS ' + validateStatus + '\n\n' +
 				'Program Info Log: ' + programLog + '\n' +
 				vertexErrors + '\n' +
 				fragmentErrors
 			);
+			if ( renderer.debug.checkShaderErrors instanceof Function ) {
```

Comment:
Callbacks in `three.js` follow a specific syntax. They start with `on....` like `onBeforeRender()` or `onBeforeCompile()`.

If @mrdoob likes the idea of a custom error callback, I would suggest to keep `checkShaderErrors` as a boolean and introduce a new callback like `renderer.debug.onShaderErrors()`.

Reply:
The triple-state is a bit cleaner, and in my mind the 'debug' category of feature sets should be succinct. But in either case I think this feature is great!